src/BoundConstantExtentLoops.cpp

```diff
@@ -46,15 +46,15 @@ class BoundLoops : public IRMutator {
     }
 
     Stmt visit(const For *op) override {
-        if (is_const(op->extent)) {
+        Expr extent = simplify(op->extent());
+        if (is_const(extent)) {
             // Nothing needs to be done
             return IRMutator::visit(op);
         }
```

Comment:
Why not take this opportunity to save the simplified constant to the `For` loop?